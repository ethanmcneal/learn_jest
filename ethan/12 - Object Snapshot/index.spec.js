import { createStore, combineReducers } from 'redux';

const usersReducer = (state, action) => {
  // Implement this reducer to pass the tests below
};

const configureStore = (initialState = {}) => {
  return createStore(
    combineReducers({
      users: usersReducer,
    }),
    initialState,
  );
};

describe('usersReducer', () => {
  it('should initialize as empty array', () => {
    const store = configureStore({});
    expect(store.getState()).toEqual({
      users: [],
    });
  });

  describe('SAVE_USER action', () => {
    // arrange
    const store = configureStore({});
    const addUserAction = user => ({
      type: 'SAVE_USER',
      user,
    });

    it('should append to state array', () => {
      // act
      store.dispatch(
        addUserAction({
          name: 'Kyle Welch',
          handle: 'kwelch',
        }),
      );

      // assert
      expect(store.getState()).toMatchSnapshot();

      // act
      store.dispatch(
        addUserAction({
          name: 'Jane Smith',
          handle: 'jsmith',
        }),
      );

      // assert
      expect(store.getState()).toMatchSnapshot();
    });

    it('should update when handle matches', () => {
      // act
      store.dispatch(
        addUserAction({
          name: 'Kyle Welch',
          handle: 'kwelch',
          role: 'Test Driven Developer',
        }),
      );

      // assert
      expect(store.getState()).toMatchSnapshot();
    });
  });
});
