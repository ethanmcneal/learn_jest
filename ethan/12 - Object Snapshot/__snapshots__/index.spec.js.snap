exports[`usersReducer SAVE_USER action should append to state array 1`] = `
Object {
  "users": Array [
    Object {
      "handle": "kwelch",
      "name": "Kyle Welch",
    },
  ],
}
`;

exports[`usersReducer SAVE_USER action should append to state array 2`] = `
Object {
  "users": Array [
    Object {
      "handle": "kwelch",
      "name": "Kyle Welch",
    },
    Object {
      "handle": "jsmith",
      "name": "Jane Smith",
    },
  ],
}
`;

exports[`usersReducer SAVE_USER action should update when handle matches 1`] = `
Object {
  "users": Array [
    Object {
      "handle": "kwelch",
      "name": "Kyle Welch",
      "role": "Test Driven Developer",
    },
    Object {
      "handle": "jsmith",
      "name": "Jane Smith",
    },
  ],
}
`;
